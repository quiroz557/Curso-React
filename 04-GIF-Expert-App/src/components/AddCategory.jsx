import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

    const [ inputValue, setInputValue ] = useState('');
    
    const handleInputChange = ({ target }) => {
        setInputValue( target.value );
    }

    const handleSubmit = ( event ) => {
        event.preventDefault();

        if( inputValue.trim().length <= 1 ) return; 

        onNewCategory( inputValue.trim() );
        setInputValue('');
    }

    return (
        <form onSubmit={ (event) => handleSubmit(event) } aria-label='form'>
          <input 
            type="text"
            placeholder='Buscar gifs' 
            value={ inputValue }
            onChange={ handleInputChange }
          /> 
        </form>
    );
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}