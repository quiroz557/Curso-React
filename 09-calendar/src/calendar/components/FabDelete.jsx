import React from 'react'
import { useCalendarStore, useUIStore } from '../../hooks'

export const FabDelete = () => {

  const { isDateModalOpen } = useUIStore();
  const { startDeletingEvent, hasEventSelected } = useCalendarStore();

  const handleClickNew = () => { 
    startDeletingEvent();
  }

  return (
    <button 
      className='btn btn-danger fab-danger'
      onClick={handleClickNew}
      style={{
        display: hasEventSelected && !isDateModalOpen ? '' : 'none',
      }}
    >
        <i className='fas fa-trash-alt'></i>
    </button>
  )
}
