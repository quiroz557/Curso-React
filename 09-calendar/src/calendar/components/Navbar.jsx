import React from 'react'
import { useAuthStore } from '../../hooks';

export const Navbar = () => {

  const { startLogout, user } = useAuthStore();

  return (
    <div className='navbar nabvar-dark bg-dark mb-4 px-4'>
        <span className='navbar-brand' style={{color:'white'}}>
            <i className='fas fa-calendar-alt'></i>
            &nbsp;
            {user.name}
        </span>

        <button 
          className='btn btn-danger'
          onClick={startLogout}
        >
            <i className='fas fa-sign-out-alt'></i>
            Salir
        </button>
    </div>
  )
}
