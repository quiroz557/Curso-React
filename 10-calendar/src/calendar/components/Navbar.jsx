import React from 'react'

export const Navbar = () => {
  return (
    <div className='navbar nabvar-dark bg-dark mb-4 px-4'>
        <span className='navbar-brand' style={{color:'white'}}>
            <i className='fas fa-calendar-alt'></i>
            &nbsp;
            Fernando
        </span>

        <button className='btn btn-danger'>
            <i className='fas fa-sign-out-alt'></i>
            Salir
        </button>
    </div>
  )
}
