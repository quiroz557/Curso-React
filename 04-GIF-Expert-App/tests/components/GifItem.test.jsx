import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

const title = 'Prueba title';
const url = 'https://media0.giphy.com/media/WOb8EeFziTQNE02WXs/giphy.gif?cid=037b74a2hmcwb6p2w4ur7qy6sdsxwc79kgp6zttlp70mk3ku&rid=giphy.gif&ct=g'

describe('Pruebas en GifItem', () => { 
    test('Debe de hacer match con el snapshot', () => { 
        const { container } = render(
            <GifItem 
                title={title} 
                url={url}
            />
        );
        expect( container ).toMatchSnapshot();
    })    

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render(
            <GifItem 
                title={title} 
                url={url}
            />
        );

        expect( screen.getByRole('img').src).toBe( url );
        expect( screen.getByRole('img').alt).toBe( title );
    })

    test('debe de mostrar el titulo en el componente', () => { 
        render(
            <GifItem 
                title={title} 
                url={url}
            />
        );
        expect( screen.getByText(title)).toBeTruthy();
    })
})