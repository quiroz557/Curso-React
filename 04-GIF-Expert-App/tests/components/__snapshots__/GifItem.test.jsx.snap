// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Prueba title"
      src="https://media0.giphy.com/media/WOb8EeFziTQNE02WXs/giphy.gif?cid=037b74a2hmcwb6p2w4ur7qy6sdsxwc79kgp6zttlp70mk3ku&rid=giphy.gif&ct=g"
    />
    <p>
      Prueba title
    </p>
  </div>
</div>
`;
